key.go

```diff
@@ -163,15 +174,16 @@ func GenerateEKeyPair(curveName string) ([]byte, GenSharedKey, error) {
 	return pubKey, done, nil
 }
 
+// StretchedKeys ...
 type StretchedKeys struct {
 	IV        []byte
 	MacKey    []byte
 	CipherKey []byte
 }
 
-// Generates a set of keys for each party by stretching the shared key.
+// KeyStretcher returns a set of keys for each party by stretching the shared key.
 // (myIV, theirIV, myCipherKey, theirCipherKey, myMACKey, theirMACKey)
-func KeyStretcher(cipherType string, hashType string, secret []byte) (StretchedKeys, StretchedKeys) {
+func KeyStretcher(cipherType string, hashType string, secret []byte) (*StretchedKeys, *StretchedKeys, error) {
```

Comment:
This is a public interface so we probably shouldn't change it (unless there's a very good reason).

Reply:
See my comment below (above?). Writes to hmac's can't fail (according to the documentation). We should probably have a comment stating this...

Reply:
My apologies! Didn't see your other comment and didn't know that about HMAC writes! TY!

I'll leave this function definition unchanged, then.

Reply:
Np, no need to apologize. This is definitely non-obvious (I had to go check the docs thinking "we can't have been *that* lazy...").